Create poolex_arctic_action.yaml

a-r-j-a-n opened the pull request:
Initial commit for the Poolex Arctic Action

a-r-j-a-n commented:
If I can assist any way in testing, please let me know (how)! :)

make-all commented:
You should be able to test by copying your added and modified files into the custom-components/tuya-local directory on your Home Assistant installation.

a-r-j-a-n commented:
I'll test it with my local version first then

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,268 @@
+            - dps_val: Heat_Intelligent
+              value: heat
+            - dps_val: Heat_Powerful
+              value: h_powerful
```
hvac_mode values must be from the predefined list https://developers.home-assistant.io/docs/core/entity/climate/#hvac-modes (the constants listed there map to all lower case strings matching the name of the constant).

In the case of multiple presets matching an hvac_mode, the full set will be accessible through preset_mode, and here all the heating values should map to heat (mark others hidden so that the  preset_mode that is selected when switching to heat mode is deterministic) and the cooling related ones to cool.

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,268 @@
+      mapping:
+        - dps_val: Heat_Intelligent
+          icon: "mdi:hot-tub"
+          value: heat
```
preset_mode can be anything, but it is recommended to take from existing translations, in this cas the translation_key pool_heatpump was already defined, and the presets defined for other pool heatpumps at https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/translations/en.json#L128 closely match (smart_heat for Heat_Intelligent, quick_heat for Heat_Powerful, and quiet_heat for Heat_Energy_Saving, and likewise for cool.  The Energy_Saving does not quite match quiet, but it seems like there is a common controller, and manufacturers choose the naming based on how they want to market it).

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,268 @@
+
+
+secondary_entities:
+  - entity: water_heater
```
New configs should not have deprecated entities.  I think you copied from the one pool heatpump that was originally created with only heating modes, and only after converting to water_heater did a user point out that it also supported cooling modes.

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,268 @@
+          - dps_val: f
+            value: Fahrenheit
+  - entity: binary_sensor
+    name: Status
```
This should be an hvac_action attribute of the climate entity, with mapping to either heating or cooling  depending on the preset mode.

make-all replied:
But actually this is dp 1, which is just the power switch. There is no need for a separate running sensor, as you can tell from the state of the climate entity whether it is switched off, or to cooling or heating. And this shouldn't be hvac_action, as it isn't giving any more info than hvac_mode does.

a-r-j-a-n replied:
Changed the configuration, based on the feedback provided. 

Also fixed the product_id, but it is still not discovered after placing it in my local custom_components/tuya_local/devices/ folder. Only matches for 83% with hydrotherm_dynamic_x8_water_heater and it doesnt come up in the type selection dropdown. Don't know how to test properly I'm afraid.

```
{
  "result": [
    {
      "active_time": 1698766405,
      "bind_space_id": "170682830",
      "category": "qn",
      "create_time": 1698766405,
      "custom_name": "Poolex",
      "icon": "smart/icon/ay1513418289516FAXSV/cfca410c1da90b02660fc721b48a951b.png",
      "id": "_redacted_",
      "ip": "_redacted_",
      "is_online": true,
      "lat": "21.7300",
      "local_key": "_redacted_",
      "lon": "15.3100",
      "model": "NC11440004",
      "name": "Heat Pumps",
      "product_id": "pzp8owd7lhtpssgj",
      "product_name": "南晟变频泳池机 ℃ ℉",
      "sub": false,
      "time_zone": "+01:00",
      "update_time": 1698766527,
      "uuid": "5734565c743afbbf"
    }
  ],
  "success": true,
  "t": 1706715008831,
  "tid": "9dcec602c04d11ee84c772fd0bbfcb23"
}
```

a-r-j-a-n commented:
Changed the configuration, based on the feedback provided.

Also fixed the product_id, but it is still not discovered after placing it in my local custom_components/tuya_local/devices/ folder. Only matches for 83% with hydrotherm_dynamic_x8_water_heater and it doesnt come up in the type selection dropdown. Don't know how to test properly I'm afraid.

```
{
  "result": [
    {
      "active_time": 1698766405,
      "bind_space_id": "170682830",
      "category": "qn",
      "create_time": 1698766405,
      "custom_name": "Poolex",
      "icon": "smart/icon/ay1513418289516FAXSV/cfca410c1da90b02660fc721b48a951b.png",
      "id": "_redacted_",
      "ip": "_redacted_",
      "is_online": true,
      "lat": "21.7300",
      "local_key": "_redacted_",
      "lon": "15.3100",
      "model": "NC11440004",
      "name": "Heat Pumps",
      "product_id": "pzp8owd7lhtpssgj",
      "product_name": "南晟变频泳池机 ℃ ℉",
      "sub": false,
      "time_zone": "+01:00",
      "update_time": 1698766527,
      "uuid": "5734565c743afbbf"
    }
  ],
  "success": true,
  "t": 1706715008831,
  "tid": "9dcec602c04d11ee84c772fd0bbfcb23"
}
```

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,267 @@
+secondary_entities:
+  - entity: climate
+    icon: "mdi:hot-tub"
+    deprecated: climate
```
In my previous comment, the intention was to suggest that you remove this deprecated entity, not change it to a second climate entity.

make-all commented on custom_components/tuya_local/devices/poolex_arctic_action.yaml:
```diff
@@ -0,0 +1,267 @@
+name: Poolex Arctic Action
```
I'm not sure why you moved the branded name here in your latest revision, this should be under products as it was before, with a generic name at top level. It is very common for pool heatpumps to be sold under an importer's brand, so essentially the same device is sold in different markets with different brands.

make-all commented:
The log message that goes along with that "83% match" should contain a list of the dps received from the device.  This should be compared with the config, and any missing dps marked as `optional: true` in the config.

a-r-j-a-n commented:
Got it working. It matches now automatically. Also added two additional entities.
